### PR TITLE
Warn about VMs

### LINKED ISSUE #323: Warn about VMs
Stolen from deepin. They throw a warning screen if you’re installing on a VM about performance etc. Probably a good idea 

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/57647282-warn-about-vms?utm_campaign=plugin&utm_content=tracker%2F46189108&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F46189108&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by meisenzahl**
Closes #323 

Shows a warning as first view if running in a virtualized environment.

# Simple solution

Bildschirmfoto von 2021-12-07 19 20 16

# Shutdown button

Bildschirmfoto von 2021-12-07 19 33 27

Maybe a bit too much with a suggested shutdown button.

### CODE REVIEW

File: src/MainWindow.vala [hunk 1]
```diff
@@ -53,8 +54,17 @@ public class Installer.MainWindow : Hdy.Window {
             margin_top = 12,
             transition_type = Gtk.StackTransitionType.SLIDE_LEFT_RIGHT
         };
+
```
Comment (meisenzahl) [inline src/MainWindow.vala:57]:
```suggestion
```

Remove unnecessary whitespace.

**Comment by meisenzahl** [comment]
@cassidyjames sorry I was not aware of #326.

Last commit changed the layout based on your PR:

Bildschirmfoto von 2021-12-07 19 50 43

**Comment by cassidyjames** [comment]
@meisenzahl shouldn't this be a part of CheckView where we perform other checks before installing?

**Comment by meisenzahl** [comment]
@cassidyjames please excuse the confusion with your PR. There is now a warning when it detects that Installer is running in a virtual machine. I took the logic almost 1:1 from your PR. In addition, I have managed to display several warnings one after the other.

https://user-images.githubusercontent.com/10796736/145271081-aaf20c33-bb17-4ea6-ba03-973307ee7a37.mp4

The checks are not executed for

- Try Demo Mode
- Custom Install

Let's try to fix this in a follow-up PR.

**Review by danrabbit — APPROVED** [review]
I can confirm that this works as intended and that checkview code looks really not fun to work, so thank you for working on this!
